lib/wasm/WebAssemblyGenerator.js

```diff
@@ -232,10 +276,41 @@ const addInitFunction = ({
 	return add(bin, [func, moduleExport, funcindex, functype]);
 };
 
+const getImportMangleMap = module => {
+	/** @type {Map<string,string>} */
+	const map = new Map();
+	let importIndex = 0;
+	for (const dep of module.dependencies) {
+		if (dep instanceof WebAssemblyImportDependency) {
+			if (dep.description.type === "GlobalType" || dep.module === null) {
+				continue;
+			}
+
+			const importedModule = dep.module;
+			const request = dep.request;
+			const exportName = dep.name;
+			const usedName = importedModule && importedModule.isUsed(exportName);
+			if (usedName !== false) {
+				map.set(
+					request + ":" + exportName,
+					Template.numberToIdentifer(importIndex++)
+				);
+			}
+		}
+	}
+	return map;
+};
+
 class WebAssemblyGenerator extends Generator {
 	generate(module) {
 		const bin = module.originalSource().source();
 
+		const initFuncId = t.identifier(
```

Comment:
Btw we need to avoid colliding with an user's export, like `__webpack_init__` is a legal export.